src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -24,10 +24,96 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "gcinfo.h"
 #include "gcinfoencoder.h"
 
+struct HWIntrinsicInfo
+{
+    instruction ins[6];          // flt, dbl, i8, i16, i32, i64
+    ssize_t     ival;
+}
+
+static const hwIntrinsicInfoArray[] = {
+#define HARDWARE_INTRINSIC(id, name, isa, ins1, ins2, ins3, ins4, ins5, ins6, ival) \
+    {ins1, ins2, ins3, ins4, ins5, ins6, ival},
+#include "hwintrinsiclistxarch.h"
+};
+
+//------------------------------------------------------------------------
+// insOfHWIntrinsic: map named intrinsic value to its corresponding instruction
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//    baseType  -- the base type of the intrinsic (flt, dbl, i8, i16, i32, i64)
+//
+// Return Value:
+//    instruction for the intrinsic
+//
+static instruction insOfHWIntrinsic(NamedIntrinsic intrinsicID, var_types baseType)
+{
+    assert(intrinsicID != NI_Illegal);
+    assert(intrinsicID > NI_HW_INTRINSIC_START && intrinsicID < NI_HW_INTRINSIC_END);
+
+    int index = -1;
+
+    switch (baseType)
+    {
+        case TYP_BYTE:
+        case TYP_UBYTE:
+            index = 2;
+            break;
+
+        case TYP_SHORT:
+        case TYP_USHORT:
+            index = 3;
+            break;
+
+        case TYP_INT:
+        case TYP_UINT:
+            index = 4;
+            break;
+
+        case TYP_LONG:
+        case TYP_ULONG:
+            index = 5;
+            break;
+
+        case TYP_FLOAT:
+            index = 0;
+            break;
+
+        case TYP_DOUBLE:
+            index = 1;
+            break;
+
+        default:
+            unreached();
+            break;
+    }
+
+    instruction ins = hwIntrinsicInfoArray[intrinsicID - NI_HW_INTRINSIC_START - 1].ins[index];
+    assert(ins != INS_invalid);
+    return ins;
+}
+
+//------------------------------------------------------------------------
+// ivalOfHWIntrinsic: map named intrinsic value to its corresponding ival
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//
+// Return Value:
+//    ival for the intrinsic
+//
+static ssize_t ivalOfHWIntrinsic(NamedIntrinsic intrinsicID)
+{
+    assert(intrinsicID != NI_Illegal);
+    assert(intrinsicID > NI_HW_INTRINSIC_START && intrinsicID < NI_HW_INTRINSIC_END);
+    return hwIntrinsicInfoArray[intrinsicID - NI_HW_INTRINSIC_START - 1].ival;
```

Comment:
note: I don't think we will have any `ival`s where the hardcoded value is `-1`, so I could probably assert that we are not returning `-1`.

Reply:
I think that all immediate values are 8 bit so it makes sense to use `int` -1 as a special value. BTW, it looks like `ival` type is `ssize_t` now, this seems unnecessary.

Reply:
Fixed.